File: URI/C#/uri-1025.cs
// URI - 1025: https://www.urionlinejudge.com.br/judge/pt/problems/view/1025

using System;

namespace uri_1025
{
    class Program
    {
        static void Main(string[] args)
        {
            int numberOfMarbles = 1, NumberOfConsultations = 1, writeLineQueryCounter = 1, queryCounter = 0,  marble = 0;
            string[] numberOfMarblesAndNumberOfConsultations = new string[] { };

            while (NumberOfConsultations != 0 && numberOfMarbles != 0)
            {
                numberOfMarblesAndNumberOfConsultations = Console.ReadLine().Split(' ');
                numberOfMarbles = int.Parse(numberOfMarblesAndNumberOfConsultations[0]);
                NumberOfConsultations = int.Parse(numberOfMarblesAndNumberOfConsultations[1]);

                int[] marbles = new int[numberOfMarbles];

                if (numberOfMarbles != 0 && NumberOfConsultations != 0)
                {
                    for (int i = 0; i < numberOfMarbles; i++)
                    {
                        marble = int.Parse(Console.ReadLine());
                        marbles[i] = marble;
                    }

                    queryCounter = 0;
                    int[] query = new int[NumberOfConsultations];

                    for (int i = 0; i < NumberOfConsultations; i++)
                    {
                        query[i] = int.Parse(Console.ReadLine());
                    }
                    Array.Sort(marbles);
                    Console.WriteLine("CASE# " + writeLineQueryCounter + ":");
                    while (queryCounter < NumberOfConsultations)
                    {
                        foundAtOrNotFound(query[queryCounter], marbles);
                        queryCounter++;
                    }
                    writeLineQueryCounter++;
                }
            }
        }

        private static void foundAtOrNotFound(int marble, int[] marbles)
        {
            int result = Array.IndexOf(marbles, marble);
            if (result < 0)
            {
                Console.WriteLine(marble + " not found");
            }
            else
            {
                result++;
                Console.WriteLine(marble + " found at " + result);
            }
        }
    }
}

File: URI/Ruby/uri-1025.rb
#URI - 1025: https://www.urionlinejudge.com.br/judge/pt/problems/view/1025

def not_have_test_case(test_case)
    return test_case[0].to_i == 0 && test_case[1].to_i == 0
end

def collect_marbles(number_of_marbles, marbles) 
    number_of_marbles.times do
        marbles << gets.to_i 
    end 
end


def collect_queries(number_of_queries, queries) 
    number_of_queries.times do 
        queries << gets.to_i
    end
end

def make_queries(queries, marbles)
    queries.each do |query| 
        result = marbles.find_index(query)
        show_result(result, query)
    end
end

def show_result(result, query) 
    if result.nil?
        puts "#{query} not found"
    else
        puts "#{query} found at #{result + 1}"
    end
end

case_counter = 1
while true
    test_case = gets.split(' ')
    if not_have_test_case(test_case)
        break
    end
    number_of_marbles = test_case[0].to_i
    number_of_queries = test_case[1].to_i
    marbles = []
    collect_marbles(number_of_marbles, marbles)
    queries = []
    collect_queries(number_of_queries, queries)
    marbles.sort!
    puts "CASE# #{case_counter}:"
    make_queries(queries, marbles)    
    case_counter += 1
end